tapiriik/services/Dropbox/dropbox.py

```diff
@@ -178,7 +178,10 @@ def cache_writeback():
                 cachedb.dropbox_cache.save(cache)
             else:
                 insert_result = cachedb.dropbox_cache.insert(cache)
-                cache["_id"] = insert_result.inserted_id
+                if hasattr(insert_result, 'inserted_id'):

```

Comment:
This was the only incompatibility I found with the latest version of things - although I only connected dropbox and runkeeper, but syncing between them worked.